packages/web3-providers-base/types.ts

```diff
@@ -1,11 +1,75 @@
+import { EventEmitter } from 'events';
+import { HttpOptions } from 'web3-providers-http/types';
+import { EthFilter, EthTransaction } from 'web3-eth/types';
+import { PrefixedHexString, ValidTypesEnum } from 'web3-utils/types';
+
+export type ProviderCallOptions = HttpOptions | undefined; // HttpOptions | WsOptions | IpcOptions
+export type RpcParams = (
+    | PrefixedHexString
+    | number
+    | EthTransaction
+    | boolean
+    | EthFilter
+)[];
+export type RpcResponseResult =
+    | string
+    | number
+    | boolean
+    | (string | number)[]
+    | { [key: string]: string | number }
+    | BigInt
+    | null
+    | any;
```

Comment:
i think either type check or allow the type to be any

Reply:
Going to create a separate PR to update the types for `web3-eth`, as merely removing `| any` caused an avalanche of type inconsistencies

Reply:
Fix by #4114